src/Tools/ILAsm/DeployILAsm.targets

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project>
+  <Choose>
+    <When Condition="'$(OS)' == 'Windows_NT'">
+      <PropertyGroup>
+        <_IlasmRid>win-x64</_IlasmRid>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))' == 'true'">
+      <PropertyGroup>
+        <_IlasmRid>osx-x64</_IlasmRid>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))' == 'true'">
+      <PropertyGroup>
+        <_IlasmRid>linux-x64</_IlasmRid>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <_IlasmRid />
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <PropertyGroup>
+    <_IlasmDeployProj>$(MSBuildThisFileDirectory)IlAsmDeploy.csproj</_IlasmDeployProj>
+    <_IlasmIncrementalBuildMarker>$(IlasmPublishDir).v$(MicrosoftNETCoreILAsmVersion)</_IlasmIncrementalBuildMarker>
+  </PropertyGroup>
+
+  <Target Name="_PublishILAsm"
+          AfterTargets="Build"
+          Inputs="$(MSBuildThisFileFullPath);$(_IlasmDeployProj)"
+          Outputs="$(_IlasmIncrementalBuildMarker)"
+          Condition="'$(TargetFramework)' == 'netcoreapp2.1'">
+
+    <Error Text="Unknown OS" Condition="'$(_IlasmRid)' == ''" />
+
+    <MSBuild Projects="$(_IlasmDeployProj)"
+             Targets="Publish"
```

Comment:
Will this implicitly restore like a `dotnet publish` would or is this the raw publish operation?

Reply:
It won't restore.